src/main/java/frc/robot/subsystems/lights/Lights.java

```diff
@@ -0,0 +1,80 @@
+package frc.robot.subsystems.lights;
+
+import com.ctre.phoenix.led.CANdle;
+import com.ctre.phoenix.led.CANdle.LEDStripType;
+import com.ctre.phoenix.led.CANdleConfiguration;
+import com.ctre.phoenix.led.RainbowAnimation;
+import com.ctre.phoenix.led.StrobeAnimation;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
+import frc.robot.RobotMap;
+import java.util.TreeMap;
+
+public class Lights {
+  private TreeMap<LightCode, Integer[]> lightOptionsMap;
+  private CANdle candle = new CANdle(RobotMap.CANDLE_CAN_ID);
+  private CANdleConfiguration config = new CANdleConfiguration();
+  private LightCode currentLightStatus = LightCode.OFF;
+
+  public enum LightCode{
+    NOTELESS, //red
+    INTAKING, //blue
+    READY_TO_SHOOT, //green
+    NO_TAG, //blinking purple
+    ALIGNING_TO_TAG, //orange
+    HOLDING_NOTE, //yellow
+    OFF, //grey
+    PARTY_MODE;
+  }
+
+  public Lights() {
+    config.stripType = LEDStripType.RGB;
+    config.brightnessScalar = 1.0;
+    candle.configAllSettings(config);
+
+    //array of integers that represents the R, G, B values of each lightcode enum
```

Comment:
this comment is not accurate as lightOptionsMap is not an array. also i would use the /** comment here */ format for the comment so that hovering over the variable lightOptionsMap will display the comment. make sure to put that comment on the line where you say private TreeMap<LightCode, Integer[]> lightOptionsMap; (line 14) though.

Reply:
Agree that we want the comment to display when hovering, but that means the comment must be above where we *create* the variable, on line 14.